### DISCUSSION TITLE
Vertically stretch and align Quasar QSelect

### DISCUSSION THREAD

**Discussion by jeff-simeon** [quasarframework/quasar#11813] [General - Components / Directives / etc]
I want my QSelects to all be vertically aligned and stretched to the same height (of the parent div)

Right now it looks like 
![image](https://user-images.githubusercontent.com/87453236/147589831-b076fdc0-00fa-4cae-bab7-f16192b461e3.png)
Which is vertically aligned but the height is not stretched on the two right QSelects.

How can I accomplish this?

I have the following component

```
<template>
  <div class="q-pa-xs">
    <div class="row">
      <q-select
        v-show="allOrgs.length > 1"
        v-model="selectedOrg"
        :bg-color="selectedOrg != null ? 'grey-2' : 'warning'"
        outlined
        dense
        use-chips
        use-input
        class="q-mr-sm q-pa-xs col"
        :options="orgs"
        label="Organization"
        @filter="(val, update) => update(() => (orgs = allOrgs.filter(i => !val || i.toLowerCase().includes(val.toLowerCase()))))"
      >
        <template #no-option>
          <q-item>
            <q-item-section class="text-grey"> No results </q-item-section>
          </q-item>
        </template></q-select
      >
      <q-select
        v-show="selectedOrg"
        v-model="selectedTenant"
        outlined
        dense
        class="q-mr-sm q-pa-xs col"
        :options="tenants"
        :bg-color="selectedTenant != null ? 'grey-2' : 'warning'"
        :loading="isLoadingTenants"
        use-chips
        use-input
        autocomplete="name"
        option-label="name"
        label="Select a tenant or enter a repository URL"
        :rules="[val => !!val]"
        @new-value="addTenant"
        @filter="(val, update) => update(() => (tenants = allTenants.filter(i => !val || i.name.toLowerCase().includes(val.toLowerCase()))))"
      >
        <template #no-option>
          <q-item>
            <q-item-section class="text-grey"> No results </q-item-section>
          </q-item>
        </template></q-select
      >
      <q-select
        v-show="selectedTenant"
        v-model="selectedVersion"
        outlined
        dense
        class="q-pa-xs col"
        :options="versions"
        :bg-color="selectedVersion != null ? 'grey-2' : 'warning'"
        :loading="isLoadingVersions"
        use-chips
        use-input
        autocomplete="label"
        map-options
        emit-value
        label="Select a tenant version ref or commit id"
        :rules="[val => !selectedTenant || !!val]"
        new-value-mode="add-unique"
        @filter="(val, update) => update(() => (versions = allVersions.filter(i => !val || i.value.toLowerCase().includes(val.toLowerCase()))))"
      >
        <template #no-option>
          <q-item>
            <q-item-section class="text-grey"> No results </q-item-section>
          </q-item>
        </template></q-select
      >
    </div>
  </div>
</template>
```

**Comment by hawkeye64** [comment]
If your parent has more height, have you tried adding class `full-height` to the q-select (or even `fit` - full height and full width)?

**Reply by jeff-simeon** [reply]
Yes, I have, without any effect. 

When I use fit and full-height, what I can see is that the container is being vertically stretched as intended but the QSelect itself is not

![image](https://user-images.githubusercontent.com/87453236/147840361-8b411479-5c74-4bfb-94ee-9f7cd64c9b57.png)

**Comment by hawkeye64** [comment]
Fyi, fit is full-height + full-width. Looking at your image out looks like it's working to me. You're letting the flex grid control the height. The q-select is the same height as the row.

**Reply by jeff-simeon** [reply]
That’s the shadow of the Chrome inspector in the image. If you zoom in closely, you’ll see that the shadow covers the row (which is correctly sized) but the QSelect does not extend vertically through the whole height of the row. 

![782E9ED1-8E8C-4B50-9D3A-1EEE38947F13](https://user-images.githubusercontent.com/87453236/147857538-ccc6d992-6fa8-4255-ad1f-f1cdcec88bc8.jpeg)

**Comment by pdanpdan** [comment]
Do you want something like this?
https://codepen.io/pdanpdan/pen/XWeEzzK

**Reply by jeff-simeon** [reply]
Yes, that's perfect except I don't want to set the min height - the height mismatch above occurs at low widths and so the left QSelect gets squished and gets a larger height. I only want the height to stretch to the parent container IF the parent container is larger than the QSelect. This doesn't seem to work in your example unless I set min-height.

**Reply by pdanpdan** [reply]
https://codepen.io/pdanpdan/pen/XWeEzzK (updated same codepen)

**Reply by jeff-simeon** [reply]
Thanks for the continued help. It's working sometimes, but still not at low widths:
![image](https://user-images.githubusercontent.com/87453236/147886531-069b9e07-18ad-4595-b32f-bb21e5b7cdeb.png)

compare with the height of the parent div

![image](https://user-images.githubusercontent.com/87453236/147886568-e07edf0f-1e0e-4a08-a5c8-3dcb1d65ac57.png)

What's interesting is that you can see, even though I have applied 100% height (even when doing manually in the debugger), it's not vertically stretching to the height of the parent div.

![image](https://user-images.githubusercontent.com/87453236/147886592-36c6804c-c836-411d-bb21-d5b4d892368d.png)

Thanks again!

**Reply by pdanpdan** [reply]
Can you please try to make a codepen that shows the problem?

**Reply by jeff-simeon** [reply]
Will do
